valhalla/sif/costfactory.h

```diff
@@ -53,9 +53,9 @@ class CostFactory {
    * @param costing    the cost type that the function creates
    * @param function   the function pointer to call to actually create the cost object
    */
-  void Register(const Costing::Type costing, factory_function_t function) {
+  void Register(const Costing::Type costing, factory_function_t&& function) {
     factory_funcs_.erase(costing);
-    factory_funcs_.emplace(costing, function);
+    factory_funcs_.emplace(costing, std::move(function));
```

Comment:
clang-tidy suggested to use `const factory_function_t&` here, but `&&` + `std::move` IMHO looks better here